torchvision/prototype/transforms/_transform.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any, Dict, Optional
+
+from torch import nn
+from torchvision.prototype.utils._internal import apply_recursively
+
+from .functional._utils import Dispatcher
+
+
+class Transform(nn.Module):
+    _DISPATCHER: Optional[Dispatcher] = None
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        return dict()
+
+    def _supports(self, obj: Any) -> bool:
+        if not self._DISPATCHER:
+            raise NotImplementedError()
+
+        return obj in self._DISPATCHER
+
+    def _dispatch(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not self._DISPATCHER:
+            raise NotImplementedError()
+
+        return self._DISPATCHER(input, **params)
+
+    def _transform_recursively(self, sample: Any, params: Dict[str, Any]) -> Any:
+        def transform(input: Any) -> Any:
+            if not self._supports(input):
+                return sample
```

Comment:
Nope, that is one key difference between the dispatchers and the transforms. Since the transforms get complete samples from the dataset, most of the transforms will not support all inputs. For example, we don't want `transforms.Normalize()` to fail if anything besides an image is passed.